File: targets/java-web/src/main/java/com/example/vulnapp/vulns/Vuln33CsvInjection.java
package com.example.vulnapp.vulns;

import org.springframework.web.bind.annotation.*;

@RestController
public class Vuln33CsvInjection {
    @GetMapping(value = "/vuln33/export", produces = "text/csv")
    public String export(@RequestParam String note) {
        // VULN:VULN-33:CWE-1236:taint user data written into CSV without neutralising formulas
        return "id,note\n1," + note + "\n";
    }
}

File: targets/kotlin-web/src/main/kotlin/com/example/vulnapp/vulns/Vuln33CsvInjection.kt
package com.example.vulnapp.vulns

import org.springframework.web.bind.annotation.*


@RestController
class Vuln33CsvInjection {

    @GetMapping("/vuln33/export", produces = ["text/csv"])
    fun export(@RequestParam note: String): String =
        // VULN:VULN-33:CWE-1236:taint user data written into CSV without neutralising formulas
        "id,note\n1,$note\n"
}
